clang/test/CAS/test-for-deterministic-outputs.c

```diff
@@ -9,6 +9,18 @@
 // CACHE-SKIPPED: remark: compile job cache skipped
 // CACHE-SKIPPED: remark: compile job cache skipped
 
+// RUN: env LLVM_CACHE_CAS_PATH=%t/cas %clang-cache \
+// RUN:   %clang -target x86_64-apple-macos11 -c %s -o %t/t.o -Rcompile-job-cache 2> %t/out.txt
+// RUN: FileCheck %s --check-prefix=CACHE-WARN --input-file=%t/out.txt
+
+/// Check still a cache miss.
+// RUN: env LLVM_CACHE_CAS_PATH=%t/cas %clang-cache \
+// RUN:   %clang -target x86_64-apple-macos11 -c %s -o %t/t.o -Rcompile-job-cache 2> %t/out.txt
+// RUN: FileCheck %s --check-prefix=CACHE-WARN --input-file=%t/out.txt
```

Comment:
We should have a test that if you define the macros yourself there is no diagnostic (which is the same behaviour as -Wdate-time).

```
-Wno-builtin-macro-redefined \
-D__DATE__='"Dec 31 1999"' \
-D__TIME__='"23:59:59" \
-D__TIMESTAMP__='"Fri Dec 31 23:59:59 1999"'
```

Reply:
It is covered already by the original test case in this file.